src/rebar3_hex_publish.erl

```diff
@@ -157,22 +157,28 @@ format_error(Reason) ->
 %% ===================================================================
 
 publish(App, HexConfig, State) ->
-    Name = rebar_app_info:name(App),
+    Publish = not(filelib:is_dir("_checkouts")),
+    case Publish of
+        true ->
+            Name = rebar_app_info:name(App),
 
-    Version = rebar_app_info:original_vsn(App),
-    ResolvedVersion = rebar_utils:vcs_vsn(App, Version, State),
-    {application, _, AppDetails} = rebar3_hex_file:update_app_src(App, ResolvedVersion),
+            Version = rebar_app_info:original_vsn(App),
+            ResolvedVersion = rebar_utils:vcs_vsn(App, Version, State),
+            {application, _, AppDetails} = rebar3_hex_file:update_app_src(App, ResolvedVersion),
 
 
-    Deps = rebar_state:get(State, {locks, default}, []),
-    {TopLevel, Excluded} = gather_deps(Deps),
+            Deps = rebar_state:get(State, {locks, default}, []),
+            {TopLevel, Excluded} = gather_deps(Deps),
 
-    case is_valid_app({App, Name, ResolvedVersion, AppDetails, Deps}) of
-        ok ->
-            publish(App, Name, ResolvedVersion, TopLevel,
-                    Excluded, AppDetails, HexConfig, State);
-        {error, Errors} ->
-            ?PRV_ERROR(Errors)
+            case is_valid_app({App, Name, ResolvedVersion, AppDetails, Deps}) of
+                ok ->
+                    publish(App, Name, ResolvedVersion, TopLevel,
+                            Excluded, AppDetails, HexConfig, State);
+                {error, Errors} ->
+                    ?PRV_ERROR(Errors)
+            end;
+        _ ->
+            ?PRV_ERROR({error, checkouts})
```

Comment:
I'm aware this is very generic, but it's also the only no-publish condition, at the moment...